src/main/java/com/google/gwt/dom/client/InputElement.java

```diff
@@ -381,6 +384,19 @@ public final void setValue(String value) {
   }-*/;
 
   /**
+   * @Mock
+   * 
+   * Set value in native element (like Selenium 'sendKeys').
+   * The new value will be available in calls to getValue().
+   * N.B. Will not trigger ValueChange events until focus lost.
+   */
+  public final void setNewValue(String value) {
+      focus();
+      setAttribute("mock.Value.new", value);
```

Comment:
In my project I do have a static call to this  'GwtMock.type(debugId, text)'.
And this function is not alone. It make the function DOMImpl.fireValueChangeIfAny actually work. 

Also there are Unit tests for this in this library. See class TestTextBoxEvents.
This is make the library usable to testing events on TextBoxBase components.
The are firing change events only on focus lost.

But agree that this function better belong in some other place in this library that we don't have yet.   Something like   BrowserMock  or GWTMockBridge. I will open a special discussion for this.

Also the function name is not perfect.  I wanted to runt this by you in hope that we will come up with more usable name and approach.

-----
Also I that believe order of Change and FosusLost events in browser may happens to be different that one we are expecting (according to spec) in application.  As described here  http://stackoverflow.com/questions/1165522/javascript-events-not-firing-in-the-expected-order
I did observed this in production and in QA of my system.

This is actually I wanted later to add more Mock to such cases to make my library more resilient ...